### PR TITLE
Fix MiddlewareQueue::insertAt

### PR DESCRIPTION
**Pull request by antograssiot**
Currently, `MiddlewareQueue::insertAt()` would work just fine with closures but can't work correctly when an instance of a Middleware is given as shown in the [book](http://book.cakephp.org/3.0/en/controllers/middleware.html#using-middleware)
It impacts also all methods that relies on it like `MiddlewareQueue::insertBefore()`

### CODE REVIEW

File: src/Http/MiddlewareQueue.php [hunk 1]
```diff
@@ -145,7 +145,7 @@ public function prepend($middleware)
      */
     public function insertAt($index, $middleware)
     {
-        array_splice($this->queue, $index, 0, $middleware);
+        array_splice($this->queue, $index, 0, [$middleware]);
```
Comment (antograssiot) [inline src/Http/MiddlewareQueue.php]: I was wondering if it is necessary to first check if `$middleware` is an array but I didn't find any cases.
Let me know if you feel it's important.

Comment (ADmad) [inline src/Http/MiddlewareQueue.php]: Nope, no check is necessary. `$middleware` will always represent a single middleware.

File: tests/test_app/TestApp/Middleware/DumbMiddleware.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @since         3.3.0
```
Comment (ADmad) [inline tests/test_app/TestApp/Middleware/DumbMiddleware.php]: 3.3.1

**Comment by ADmad** [comment]
Good find @antograssiot

**Pull request merged**